cpp/dolfinx/fem/Form.h

```diff
@@ -606,9 +606,18 @@ class Form
   // True if permutation data needs to be passed into these integrals
   bool _needs_facet_permutations;
 
-  // Entity maps (see Form documentation)
-  std::map<std::shared_ptr<const mesh::Mesh<geometry_type>>,
-           std::vector<std::int32_t>>
-      _entity_maps;
+  // Domain data for argument functions
+  //  [rank_i][IntegralType, integral(id), cell_type] -> entity map
+  std::vector<std::map<
```

Comment:
I'm think it's fine with storing these here, rather than just the entitiy maps. But, this approach might use more memory than before for some integrals

Reply:
Any memory change would be zero to minimal because the data can be shared across integrals. The re-computation cost is avoided. Some more work would allow more data sharing for more complex cases.

Major change is avoiding need to store shared pointer to meshes, which was ugly.